Zend/zend_builtin_functions.c

```diff
@@ -637,26 +637,21 @@ ZEND_FUNCTION(get_called_class)
    Retrieves the parent class name for object or class or current scope or false if not in a scope. */
 ZEND_FUNCTION(get_parent_class)
 {
-	zval *arg;
+	zend_string *str = NULL;
+	zend_object *object = NULL;
 	zend_class_entry *ce = NULL;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|z", &arg) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(0, 1)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR_OR_OBJ(str, object)
```

Comment:
@nikic In the newest implementation, I do not use `Z_PARAM_STR_OR_OBJ()` anymore. Hopefully we will be able to deprecate calling this function with no argument.